pkg/expression/builtin_string.go

```diff
@@ -758,6 +758,7 @@ func (c *reverseFunctionClass) getFunction(ctx BuildContext, args []Expression)
 	if err := c.verifyArgs(args); err != nil {
 		return nil, err
 	}
+	bitTypeFlag := types.IsTypeBit(args[0].GetType())
```

Comment:
Sorry. I have another proposal 🤦 . It turns out that the type implicit conversion is really a chaos. We have more issues for `BIT`, see: https://github.com/pingcap/tidb/issues/50855 and https://github.com/pingcap/tidb/issues/50850 .

The problem is that: implicitly casting a `BIT` value to string should give `binary` charset/collation, and explicitly selecting `cast(bit_col as char)` will produce a normal string with default collation/charset. Given these information, I proposed to modify `WrapWithCastAsString` by adding the following lines:

```go

	// Implicitly casting BIT to string will make it a binary
	if exprTp.GetType() == mysql.TypeBit {
		tp.SetCharset(charset.CharsetBin)
		tp.SetCollate(charset.CollationBin)
	}
```

With this modification, the `bitTypeFlag` here is not needed and it may also fix https://github.com/pingcap/tidb/issues/50855 and https://github.com/pingcap/tidb/issues/50850 🤔 (I'm not sure). If so, please add some tests for them.

Also cc @zanmato1984 PTAL

Reply:
Get, i will test it as you suggestions.

Reply:
> here is not needed and it may also fix #50855 and #50850 🤔 (I'm not sure). If so, please add some tests for them.

@YangKeao The results is as you expected, I tested it can solve the other issure too.